pkg/config/repoconfig/yaml.go

```diff
@@ -0,0 +1,34 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package repoconfig
+
+const (
+	defaultApprovals = 2
+	defaultLabel     = "ready-to-test"
+	filename         = ".submarinerbot.yaml"
+)
+
+type BotConfigYAML struct {
+	LabelApproved *LabelApprovedYAML `yaml:"label-approved"`
+}
+
+type LabelApprovedYAML struct {
+	Approvals *int
+	Label     *string
+}
```

Comment:
I understand why you're doing this, but it seems kind of over-engineered to me.

Reply:
I think it's an ok compromise to be able to process the pointers before any usage.